sdk/core/azure-core/test/ut/curl_connection_pool.cpp

```diff
@@ -0,0 +1,172 @@
+      std::string const CAinfo = "someFakePath";
+      std::string const secondExpectedKey = "httpbin.org" + CAinfo + "011";
+      {
+        // Creating a new connection with default options
+        Azure::Core::Http::CurlTransportOptions options;
+        options.CAInfo = CAinfo;
+        auto connection = Azure::Core::Http::CurlConnectionPool::GetCurlConnection(req, options);
+        EXPECT_EQ(connection->GetConnectionKey(), secondExpectedKey);
+        // One connection still in the pool after getting a new connection and with first expected
+        // key
+        EXPECT_EQ(Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.size(), 1);
+        EXPECT_EQ(
+            Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.begin()
+                ->second.begin()
+                ->get()
+                ->GetConnectionKey(),
+            expectedConnectionKey);
+
+        auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+            req, std::move(connection), options.HttpKeepAlive);
+        // Simulate connection was used already
+        session->m_lastStatusCode = Azure::Core::Http::HttpStatusCode::Ok;
+        session->m_sessionState = Azure::Core::Http::CurlSession::SessionState::STREAMING;
+      }
+
+      // Now there should be 2 index wit one connection each
+      EXPECT_EQ(Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.size(), 2);
+      values = Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.begin();
+      EXPECT_EQ(values->second.size(), 1);
+      EXPECT_EQ(values->second.begin()->get()->GetConnectionKey(), expectedConnectionKey);
+      values++;
+      EXPECT_EQ(values->second.size(), 1);
+      EXPECT_EQ(values->second.begin()->get()->GetConnectionKey(), secondExpectedKey);
+
+      // Test re-using same custom config
+      {
+        // Creating a new connection with default options
+        Azure::Core::Http::CurlTransportOptions options;
+        options.CAInfo = CAinfo;
+        auto connection = Azure::Core::Http::CurlConnectionPool::GetCurlConnection(req, options);
+        EXPECT_EQ(connection->GetConnectionKey(), secondExpectedKey);
+        // One connection still in the pool after getting a new connection and with first expected
+        // key
+        EXPECT_EQ(Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.size(), 1);
+        EXPECT_EQ(
+            Azure::Core::Http::CurlConnectionPool::ConnectionPoolIndex.begin()
+                ->second.begin()
+                ->get()
+                ->GetConnectionKey(),
+            expectedConnectionKey);
+
+        auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+            req, std::move(connection), options.HttpKeepAlive);
+        // Simulate connection was used already
+        session->m_lastStatusCode = Azure::Core::Http::HttpStatusCode::Ok;
+        session->m_sessionState = Azure::Core::Http::CurlSession::SessionState::STREAMING;
```

Comment:
nit: this test has a bit of code duplication that you might be able to extract into helpers.